app/models/course/assessment/answer/programming.rb

```diff
@@ -56,9 +56,20 @@ def download(dir)
   def assign_params(params)
     acting_as.assign_params(params)
 
+    params[:import_files]&.each do |file|
+      files.build(
+        filename: file.original_filename,
+        content: File.read(file.tempfile.path)
+      )
+    end
+
     params[:files_attributes]&.each do |file_attributes|
       file = files.find { |f| f.id == file_attributes[:id].to_i }
-      file.content = file_attributes[:content]
+      if file_attributes[:_destroy]
+        files.delete(file)
```

Comment:
Actually I thought this part would be automatically done via `accepts_nested_attributes_for :files, ...`, is this set of logic required?

http://api.rubyonrails.org/classes/ActiveRecord/NestedAttributes/ClassMethods.html